docs/sphinx/cookbook/file_allocation.rst

```diff
@@ -0,0 +1,114 @@
+.. _file_allocation:
+
+==================================
+Using File System Allocator (FILE)
+==================================
+
+Umpire supports the use of file based memory allocation. When ``ENABLE_FILE_RESOURCE`` 
+is enabled, the environment variables ``UMPIRE_MEMORY_FILE_DIR`` can be used to determine 
+where memory can be allocated from:
+
+  ======================   ======================   =======================================================
+  Variable                 Default                  Description
+  ======================   ======================   =======================================================
+  UMPIRE_MEMORY_FILE_DIR   ./                       Directory to create and allocate file based allocations
+  ======================   ======================   =======================================================
+
+Requesting the allocation takes two steps: 1) getting a "FILE" allocator, 
+2) requesting the amount of memory to allocate.
+
+.. literalinclude:: ../../../examples/cookbook/recipe_filesystem_memory_allocation.cpp
+                    :lines: 12-15
+
+To deallocate:
+
+.. literalinclude:: ../../../examples/cookbook/recipe_filesystem_memory_allocation.cpp
+                    :lines: 17
+
+The complete example is included below:
+
+.. literalinclude:: ../../../examples/cookbook/recipe_filesystem_memory_allocation.cpp
+
+=============================
+Using Burst Buffers On Lassen
+=============================
+
+On Lassen, 1) Download the latest version of Umpire 2) request a private node to build:
+
+.. code-block:: bash
+
+  $ git clone --recursive https://github.com/LLNL/Umpire.git
+  $ lalloc 1 -stage storage=64
```

Comment:
I think you should add a comment on the `-state storage=64` argument (since that's required for the burst buffers)